process_data.go

```diff
@@ -55,8 +55,8 @@ func (dsp *DataStreamProcessor) SetProjectorsBasis(projectors mat.Dense, basis m
 	if brows != dsp.NSamples {
 		return fmt.Errorf("basis has wrong size, has rows: %v, want: %v", brows, dsp.NSamples)
 	}
-	dsp.projectors = projectors
-	dsp.basis = basis
+	dsp.projectors = *projectors
+	dsp.basis = *basis
```

Comment:
Make `dsp.basis` and `dsp.projectors` hold `*mat.Dense` and these derefs will go away.

Reply:
that makes sense. I didn't do that because I later use `IsZero` and I couldn't figure out how to initialize them without calling `NewDense` which requires meaningless (since I don't have it yet) size information?

Reply:
I guess I could do `var temp mat.Dense;dsp.projectors=&temp`?

Reply:
Yes, that's how I would do it.